NewForce-at-Mountwest/bangazon-api-radvioli
# Db product resource

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Created the resources for Product and Product Type. These resources will contain the Products in which Customers will Buy and Sell as well as the Product Type which is the category for each Product that allows for future search filtering and better organization. 
The Product resource is imperative for Bangazon because without them the Customers would not be able to use Bangazon for its intended purposes. The Product Type resource is less imperative, but still important for ease-of-access and usability.


## Related Ticket(s)
Ticket #2: Allow developers to access Product resource
Ticket #5: Allow developers to access Product Type resource

## Steps to Test

1. git add and git commit whatever you're working on on your branch
2. git checkout master to switch to your local master branch
3. git fetch --all will give you access to your teammate's branches (if they've pushed them up)
4. git checkout <your-teammate's-branch-name> to switch to their branch
5. If you don't see the most recent version of their files, try git pull origin <your-teammate's-branch-name>
6. Run your server on their branch and make sure everything looks okay

Once server is running, use your browser to navigate to api/products to ensure that correct information is displayed.

After that step, use your browser to navigate to api/producttypes to ensure that correct information is displayed.

Afterwards, in Visual Studio, go to the TestProduct.cs file under TestBangazonAPI and right-click on it and click on Run Tests. Ensure that all tests come back green.

Then, go to the TestProductType.cs file under TestBangazonAPI and right-click on it and click on Run Tests. Ensure that all tests come back green.

Check verbiage and comments to make sure they are legible and understandable.

# Checklist:
- [ ✔️ ] I have commented my code, particularly in hard-to-understand areas
- [ ✔️ ] I have made corresponding changes to the documentation
- [ ✔️  ] My changes generate no new errors
- [ ✔️  ] I have added tests that prove my fix is effective or that my feature works
- [ ✔️  ] New and existing unit tests pass locally with my changes
